contracts/Aavegotchi/facets/AavegotchiLendingFacet.sol

```diff
@@ -0,0 +1,277 @@
+    ///@param _length How many aavegotchi rentals to return
+    ///@return rentals_ An array of rental
+    function getOwnerAavegotchiRentals(address _owner, uint256 _length) external view returns (AavegotchiRental[] memory rentals_) {
+        uint256 rentalId = s.aavegotchiOwnerRentalHead[_owner];
+        rentals_ = new AavegotchiRental[](_length);
+        uint256 listIndex;
+        for (; rentalId != 0 && listIndex < _length; listIndex++) {
+            rentals_[listIndex] = s.aavegotchiRentals[rentalId];
+            rentalId = s.aavegotchiOwnerRentalListItem[rentalId].childRentalId;
+        }
+        assembly {
+            mstore(rentals_, listIndex)
+        }
+    }
+
+    ///@notice Query a certain amount of aavegotchi rentals
+    ///@param _length How many rentals to return
+    ///@return rentals_ An array of rental
+    function getAavegotchiRentals(uint256 _length) external view returns (AavegotchiRental[] memory rentals_) {
+        uint256 rentalId = s.aavegotchiRentalHead;
+        rentals_ = new AavegotchiRental[](_length);
+        uint256 listIndex;
+        for (; rentalId != 0 && listIndex < _length; listIndex++) {
+            rentals_[listIndex] = s.aavegotchiRentals[rentalId];
+            rentalId = s.aavegotchiRentalListItem[rentalId].childRentalId;
+        }
+        assembly {
+            mstore(rentals_, listIndex)
+        }
+    }
+
+    function isAavegotchiLent(uint256 _erc721TokenId) external view returns (bool) {
+        return LibAavegotchiLending.isAavegotchiLent(_erc721TokenId);
+    }
+
+    ///@notice Allow an original aavegotchi owner to add request for rental
+    ///@dev If the rental request exist, cancel it and replaces it with the new one
+    ///@dev If the rental is active, unable to cancel
+    ///@param _erc721TokenId The identifier of the NFT to rent
+    ///@param _initialCost The rental fee of the aavegotchi in $GHST
+    ///@param _period The rental period of the aavegotchi, unit: second
+    ///@param _revenueSplit The revenue split of the rental, 3 values, sum of the should be 100
+    ///@param _receiver The 3rd account for receive revenue split, can be address(0)
+    ///@param _whitelistId The identifier of whitelist for agree rental, if 0, allow everyone
+    function addAavegotchiRental(
+        uint256 _erc721TokenId,
+        uint256 _initialCost,
+        uint256 _period,
+        uint256[3] calldata _revenueSplit,
+        address _receiver,
+        uint256 _whitelistId,
+        address[] calldata _includes
+    ) external {
+        IERC721 erc721Token = IERC721(address(this));
+        address sender = LibMeta.msgSender();
+        require(erc721Token.ownerOf(_erc721TokenId) == sender, "AavegotchiLending: Not owner of aavegotchi");
+        require(
+            erc721Token.isApprovedForAll(sender, address(this)) || erc721Token.getApproved(_erc721TokenId) == address(this),
+            "AavegotchiLending: Not approved for transfer"
+        );
```

Comment:
Need to add `_erc721TokenAddress == address(this) ||` because Aavegotchis should be approved by default